lib/dependabot/file_updaters/java_script/npm_and_yarn/dependency_files_filterer.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require "dependabot/file_updaters/java_script/npm_and_yarn"
+
+module Dependabot
+  module FileUpdaters
+    module JavaScript
+      class NpmAndYarn
+        class DependencyFilesFilterer
+          def self.filter(dependency_files:, dependencies:)
```

Comment:
Switch to an `initialize` and `filtered_files` pattern (more consistent with other classes and means you can have private methods).